libs/client/WebhookClient.lua

```
@@ -0,0 +1,314 @@
+	if webhook then
+		for key, value in pairs(webhook) do
+			if type(value) ~= "table" then
+				self._options[key] = value
+			else 
+				for user_key, user_value in pairs(value) do
+					if not self._options[key] then self._options[key] = {} end
+					self._options[key]["_" .. user_key] = user_value
+				end
+			end
+		end
+	end
+
+	if options then
+		modify(self, options)
+		for k, v in pairs(options) do
+			self._options[k] = v
+		end
+	end
+	self._webhook = Webhook(self._options, self, self._api)
+end
+
+for name, level in pairs(logLevel) do
+	WebhookClient[name] = function(self, fmt, ...)
+		local msg = self._logger:log(level, fmt, ...)
+		if #self._listeners[name] > 0 then
+			return self:emit(name, msg or format(fmt, ...))
+		end
+	end
+end
+
+-- Getters
+function get.avatarURL(self, size, ext)
+	return self._options.avatarURL ~= "" and self._options.avatarURL or resolveImage(self._options.avatar, self._id)
+end
+
+function get.channelId(self)
+	return self._options.channel_id
+end
+
+function get.guildId(self)
+	return self._webhook.guildId
+end
+
+function get.id(self)
+	return self._webhook.id
+end
+
+function get.name(self)
+	return self._options.name
+end
+
+function get.token(self)
+	return self._webhook.token
+end
+
+function get.user(self)
+	return clean(self._webhook.user) -- does not return a full user object because it is not linked with Client
+end
+
+
+-- Setters and Modifiers
+function WebhookClient:setAvatar(avatar)
+	self._options.avatarURL = avatar
+	avatar = resolveImage(avatar) or Resolver.base64(avatar) or null
+
+	local bool, err = self:modify({avatar = avatar})
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:setChannelId(channelId)
+	self._options.channelId = channelId
+	local bool, err = self:modify({channel_id = channelId})
+
+	if bool then
+		return self
+	else
+		self:error(err)
```

Should return the error instead of logging it.